subprojects/core-api/src/main/java/org/gradle/api/model/ObjectFactory.java

```diff
@@ -212,6 +211,7 @@ public interface ObjectFactory {
      * @return The property. Never returns null.
      * @since 4.3
      */
+    @ManagedObjectCreator
```

Comment:
There was a comment about needing to have these annotations on public methods. It may be possible to put `ManagedObjectCreator` on the implementation methods if we want. But I think we still need `ManagedObjectProvider` on the public type given the way these services are discovered.

However, I do think this serves as a good documentation for the user. It pretty directly declares that this method is the one that will eventually go and create the `ManagedObject` you want to instantiate.